### ISSUE TITLE
how to disable whipser transcription

### ISSUE THREAD

**Issue by suisuyy** [openai/openai-realtime-api-beta#15]
I see this in README:
```
client.updateSession({
  turn_detection: { type: 'none' }, // or 'server_vad'
  input_audio_transcription: { model: 'whisper-1' },
});
```
there is "model: 'whisper-1'", so it still use wisper model to transcribe, can we disable whisper , make gpt4o listen user voice directly or use gpt4o for transcribing

**Comment by suisuyy** [comment]
and in [lib/client.js](https://github.com/openai/openai-realtime-api-beta/blob/0126e4bfc19901598c3f20d0a4b32bb3e0bea376/lib/client.js#L210)
it set transcriptionModels to whisper-1, but no option to change 

```
 constructor({ url, apiKey, dangerouslyAllowAPIKeyInBrowser, debug } = {}) {
    super();
    this.defaultSessionConfig = {
      modalities: ['text', 'audio'],
      instructions: '',
      voice: 'alloy',
      input_audio_format: 'pcm16',
      output_audio_format: 'pcm16',
      input_audio_transcription: null,
      turn_detection: null,
      tools: [],
      tool_choice: 'auto',
      temperature: 0.8,
      max_response_output_tokens: 4096,
    };
    this.sessionConfig = {};
    this.transcriptionModels = [
      {
        model: 'whisper-1',
      },
    ];

```

**Comment by Stevenic** [comment]
> there is "model: 'whisper-1'", so it still use whisper model to transcribe, can we disable whisper , make gpt4o listen user voice directly or use gpt4o for transcribing

I'm pretty sure they're just using whisper-1 to do text transcription. The text transcription is needed to make conversation history work so you wouldn't want to disable it even if you could.  You also wouldn't want to use gpt-4o to do the transcription because a) it hasn't been trained for transcription and b) it would be significantly more expensive (both latency and cost wise.)

**Comment by suisuyy** [comment]
> I'm pretty sure they're just using whisper-1 to do text transcription.
Thanks, GPT-4O supports voice in and voice out, so is it possible to input voice to GPT-4O directly and get voice response, or is the capability limited?

**Comment by Stevenic** [comment]
> Thanks, GPT-4O supports voice in and voice out, so is it possible to input voice to GPT-4O directly and get voice response, or is the capability limited?

They're using whisper to get a "text" based transcription of the "audio" that was generated by the gpt-4o call.  Models don't currently now how to output both text and audio simultaneously so you have no choice but to do this as a 2 step process. You have to first generate the "audio" and then you can generate the "text" of the response using transcription.

What I'm saying is that for that second step you're better off using a cheaper model like "whisper" that's specifically trained to do transcription. While you could use a second call to gpt-4o to do that task it would cost more, take longer, and might have errors because it's not specifically trained to do transcription.

**Comment by suisuyy** [comment]
> What I'm saying is that for that second step you're better off using a cheaper model like "whisper" that's specifically trained to do transcription. While you could use a second call to gpt-4o to do that task it would cost more, take longer, and might have errors because it's not specifically trained to do transcription.

Yeah, but how it understand user's voice,  do you think they use Whisper to transcribe user's voice and then give the transcription to GPT-4o, I have asked the model, it admitted it does things like this.
![image](https://github.com/user-attachments/assets/dc6cb9c1-ead8-4171-b5ba-316f200c0c33)

**Comment by stippi** [comment]
I _assume_ this is how it works: The GPT-4o model is getting the audio from the user and outputting audio itself. I parallel, the whisper-1 model is used for transcription. This is only necessary so you get the transcribed text in addition to the audio in the events. Seeing that the whisper-1 transcription is optional, it makes no sense to assume it is required and actually used for the GPT-4o input.

**Comment by suisuyy** [comment]
> I _assume_ this is how it works: The GPT-4o model is getting the audio from the user and outputting audio itself. I parallel, the whisper-1 model is used for transcription. This is only necessary so you get the transcribed text in addition to the audio in the events. Seeing that the whisper-1 transcription is optional, it makes no sense to assume it is required and actually used for the GPT-4o input.
> 
> The model output in your screenshot might well be a hallucination. I had cases where the transcription made no sense, but the model understood what I was saying regardless.

thanks for explaining, i hope so.
because the model can not  detect my language pronoucation errors, even i speak 'dog' like 'darg', it still think i speak 'dog' and pronoucation is correct . so i suspect it use transcription to understand user's voice

**Comment by guvenkaranfil** [comment]
Hello @suisuyy ,
We are having the same isssue. Have you find a solution?
